### ISSUE TITLE
Multiprocessing in Spyder 5.4.3

### ISSUE THREAD

**Issue by BrazilForever11** [spyder-ide/spyder#21106]
Here is content of the module square_function.py:

```
def square_function(x):
    return x*x
```

Here is what I run interactively in Spyder: 

```
from multiprocessing import Pool
import square_function

if __name__=='__main__':
    with Pool(5) as p:
            print(p.map(square_function.square_function, [1, 2, 3]))

```

I also tried to run: 

```
from multiprocessing import Pool
import square_function

with Pool(5) as p:
    print(p.map(square_function.square_function, [1, 2, 3]))
```

It hangs and does not generate any output. The same code runs fine in Jupyter Notebook

Here are the details:

Spyder: 5.4.3
Python: 3.9.12
Windows 10

**Comment by dalthviz** [comment]
Regarding running the code without defining `__main__` , you can run it using the `Run selection or current line` to prevent it getting stuck:

![multiprocessing_selection](https://github.com/spyder-ide/spyder/assets/16781833/90446309-1834-4952-81de-2089fb32e0bd)

So I was able to run both versions of the code without issue in the following env with Python 3.9 and Spyder 5.4.4:

<details>

```
# packages in environment at C:\Users\dalth\anaconda3\envs\spyder-cf-39:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.13             pyhd8ed1ab_0    conda-forge
arrow                     1.2.3              pyhd8ed1ab_0    conda-forge
astroid                   2.15.6           py39hcbf5309_0    conda-forge
asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
atomicwrites              1.4.1              pyhd8ed1ab_0    conda-forge
attrs                     23.1.0             pyh71513ae_1    conda-forge
autopep8                  2.0.2              pyhd8ed1ab_0    conda-forge
babel                     2.12.1             pyhd8ed1ab_1    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
bcrypt                    3.2.2            py39ha55989b_1    conda-forge
beautifulsoup4            4.12.2             pyha770c72_0    conda-forge
binaryornot               0.4.4                      py_1    conda-forge
black                     23.7.0           py39hcbf5309_1    conda-forge
bleach                    6.0.0              pyhd8ed1ab_0    conda-forge
brotli-python             1.0.9            py39h99910a6_9    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2023.5.7             h56e8100_0    conda-forge
certifi                   2023.5.7           pyhd8ed1ab_0    conda-forge
cffi                      1.15.1           py39h68f70e3_3    conda-forge
chardet                   5.1.0            py39hcbf5309_0    conda-forge
charset-normalizer        3.2.0              pyhd8ed1ab_0    conda-forge
click                     8.1.6           win_pyh7428d3b_0    conda-forge
cloudpickle               2.2.1              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.1.3              pyhd8ed1ab_0    conda-forge
cookiecutter              2.2.3              pyh1a96a4e_0    conda-forge
cryptography              41.0.2           py39hb6bd5e6_0    conda-forge
debugpy                   1.6.7            py39h99910a6_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
diff-match-patch          20230430           pyhd8ed1ab_0    conda-forge
dill                      0.3.6              pyhd8ed1ab_1    conda-forge
docstring-to-markdown     0.12               pyhd8ed1ab_0    conda-forge
docutils                  0.20.1           py39hcbf5309_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
flake8                    6.0.0              pyhd8ed1ab_0    conda-forge
gettext                   0.21.1               h5728263_0    conda-forge
glib                      2.76.4               h12be248_0    conda-forge
glib-tools                2.76.4               h12be248_0    conda-forge
gst-plugins-base          1.22.4               h001b923_1    conda-forge
gstreamer                 1.22.4               hb4038d2_1    conda-forge
icu                       72.1                 h63175ca_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
importlib-metadata        6.8.0              pyha770c72_0    conda-forge
importlib_metadata        6.8.0                hd8ed1ab_0    conda-forge
importlib_resources       6.0.0              pyhd8ed1ab_1    conda-forge
inflection                0.5.1              pyh9f0ad1d_0    conda-forge
intervaltree              3.1.0              pyhd8ed1ab_1    conda-forge
ipykernel                 6.24.0             pyh6817e22_0    conda-forge
ipython                   8.14.0             pyh08f2357_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isort                     5.12.0             pyhd8ed1ab_1    conda-forge
jaraco.classes            3.3.0              pyhd8ed1ab_0    conda-forge
jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
jellyfish                 1.0.0            py39hf21820d_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
jsonschema                4.18.4             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2023.7.1           pyhd8ed1ab_0    conda-forge
jupyter_client            8.3.0              pyhd8ed1ab_0    conda-forge
jupyter_core              5.3.1            py39hcbf5309_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
keyring                   24.2.0           py39hcbf5309_0    conda-forge
krb5                      1.20.1               heb0366b_0    conda-forge
lazy-object-proxy         1.9.0            py39ha55989b_0    conda-forge
libclang                  16.0.6          default_heb8d277_0    conda-forge
libclang13                16.0.6          default_hc80b9e7_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.76.4               he8f3873_0    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
libjpeg-turbo             2.1.5.1              hcfcfb64_0    conda-forge
libogg                    1.3.4                h8ffe710_1    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libspatialindex           1.9.3                h39d44d4_4    conda-forge
libsqlite                 3.42.0               hcfcfb64_0    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp                   1.3.1                hcfcfb64_0    conda-forge
libwebp-base              1.3.1                hcfcfb64_0    conda-forge
libxml2                   2.11.4               hc3477c8_0    conda-forge
libzlib                   1.2.13               hcfcfb64_5    conda-forge
markupsafe                2.1.3            py39ha55989b_0    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mccabe                    0.7.0              pyhd8ed1ab_0    conda-forge
mistune                   3.0.0              pyhd8ed1ab_0    conda-forge
more-itertools            9.1.0              pyhd8ed1ab_0    conda-forge
mypy_extensions           1.0.0              pyha770c72_0    conda-forge
nbclient                  0.8.0              pyhd8ed1ab_0    conda-forge
nbconvert                 7.7.1              pyhd8ed1ab_1    conda-forge
nbconvert-core            7.7.1              pyhd8ed1ab_1    conda-forge
nbconvert-pandoc          7.7.1              pyhd8ed1ab_1    conda-forge
nbformat                  5.9.1              pyhd8ed1ab_0    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
numpydoc                  1.5.0              pyhd8ed1ab_0    conda-forge
openssl                   3.1.1                hcfcfb64_1    conda-forge
packaging                 23.1               pyhd8ed1ab_0    conda-forge
pandoc                    3.1.3                h57928b3_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
paramiko                  3.2.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pathspec                  0.11.1             pyhd8ed1ab_0    conda-forge
pcre2                     10.40                h17e33f8_0    conda-forge
pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pip                       23.2               pyhd8ed1ab_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
platformdirs              3.9.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.2.0              pyhd8ed1ab_0    conda-forge
ply                       3.11                       py_1    conda-forge
prompt-toolkit            3.0.39             pyha770c72_0    conda-forge
prompt_toolkit            3.0.39               hd8ed1ab_0    conda-forge
psutil                    5.9.5            py39ha55989b_0    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pycodestyle               2.10.0             pyhd8ed1ab_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydocstyle                6.3.0              pyhd8ed1ab_0    conda-forge
pyflakes                  3.0.1              pyhd8ed1ab_0    conda-forge
pygments                  2.15.1             pyhd8ed1ab_0    conda-forge
pylint                    2.17.4             pyhd8ed1ab_0    conda-forge
pylint-venv               3.0.2              pyhd8ed1ab_0    conda-forge
pyls-spyder               0.4.0              pyhd8ed1ab_0    conda-forge
pynacl                    1.5.0            py39h09fa780_2    conda-forge
pyqt                      5.15.7           py39hb77abff_3    conda-forge
pyqt5-sip                 12.11.0          py39h99910a6_3    conda-forge
pyqtwebengine             5.15.7           py39h2f4a3f1_3    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
python                    3.9.16          h4de0772_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.17.1             pyhd8ed1ab_0    conda-forge
python-lsp-black          1.3.0              pyhd8ed1ab_0    conda-forge
python-lsp-jsonrpc        1.0.0              pyhd8ed1ab_0    conda-forge
python-lsp-server         1.7.4              pyhd8ed1ab_0    conda-forge
python-lsp-server-base    1.7.4              pyhd8ed1ab_0    conda-forge
python-slugify            8.0.1              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      3_cp39    conda-forge
pytoolconfig              1.2.5              pyhd8ed1ab_0    conda-forge
pytz                      2023.3             pyhd8ed1ab_0    conda-forge
pywin32                   304              py39h99910a6_2    conda-forge
pywin32-ctypes            0.2.2            py39hcbf5309_0    conda-forge
pyyaml                    6.0              py39ha55989b_5    conda-forge
pyzmq                     25.1.0           py39hea35a22_0    conda-forge
qdarkstyle                3.1                pyhd8ed1ab_0    conda-forge
qstylizer                 0.2.2              pyhd8ed1ab_0    conda-forge
qt-main                   5.15.8              h2c8576c_12    conda-forge
qt-webengine              5.15.8               h5b1ea0b_0    conda-forge
qtawesome                 1.2.3              pyhd8ed1ab_0    conda-forge
qtconsole                 5.4.3              pyhd8ed1ab_0    conda-forge
qtconsole-base            5.4.3              pyha770c72_0    conda-forge
qtpy                      2.3.1              pyhd8ed1ab_0    conda-forge
referencing               0.30.0             pyhd8ed1ab_0    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
rope                      1.9.0              pyhd8ed1ab_0    conda-forge
rpds-py                   0.9.2            py39hf21820d_0    conda-forge
rtree                     1.0.1            py39h09fdee3_1    conda-forge
setuptools                68.0.0             pyhd8ed1ab_0    conda-forge
sip                       6.7.9            py39h99910a6_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
sphinx                    7.0.1              pyhd8ed1ab_0    conda-forge
sphinxcontrib-applehelp   1.0.4              pyhd8ed1ab_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    2.0.1              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_2    conda-forge
spyder                    5.4.4            py39hcbf5309_0    conda-forge
spyder-kernels            2.4.4           win_pyh7428d3b_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
text-unidecode            1.3                        py_0    conda-forge
textdistance              4.5.0              pyhd8ed1ab_0    conda-forge
three-merge               0.1.1              pyh9f0ad1d_0    conda-forge
tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h8ffe710_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tomlkit                   0.11.8             pyha770c72_0    conda-forge
tornado                   6.3.2            py39ha55989b_0    conda-forge
traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
typing-extensions         4.7.1                hd8ed1ab_0    conda-forge
typing_extensions         4.7.1              pyha770c72_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
ujson                     5.7.0            py39h99910a6_0    conda-forge
unidecode                 1.3.6              pyhd8ed1ab_0    conda-forge
urllib3                   2.0.4              pyhd8ed1ab_0    conda-forge
vc                        14.3                h64f974e_17    conda-forge
vc14_runtime              14.36.32532         hfdfe4a8_17    conda-forge
vs2015_runtime            14.36.32532         h05e6639_17    conda-forge
watchdog                  3.0.0            py39hcbf5309_0    conda-forge
wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
whatthepatch              1.0.5              pyhd8ed1ab_0    conda-forge
wheel                     0.40.0             pyhd8ed1ab_1    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
wrapt                     1.15.0           py39ha55989b_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
yapf                      0.33.0             pyhd8ed1ab_1    conda-forge
zeromq                    4.3.4                h0e60522_1    conda-forge
zipp                      3.16.2             pyhd8ed1ab_0    conda-forge
zstd                      1.5.2                h12be248_7    conda-forge
```

</details>

@BrazilForever11 do you have Spyder installed from the default anaconda channel? Could it be possible for you to create a new conda environment to get the latest Spyder release and check if the error persist? For that you can run from an Anaconda Prompt something like:

    conda create -n spyder-cf -c conda-forge python=3.9 spyder

Let us know!

**Comment by BrazilForever11** [comment]
Thank you so much! I did what you said. Afterwards I activated `spyder-cf` environment and typed and ran `spyder`.  

Now I have dual behaviour. 
First scenario:
Immediately after launching Spyder I select my code and click "Run selection or current Line (Ctrl+Return)" - nothing happens. Spyder freezes. I shut down Spyder through task manager.

Second scenario:
Immediately after launching Spyder I select my code and click "Run file (F5)". Everything works. I repeat with "Run selection or current Line (Ctrl+Return)" and it also works. 

So somehow running whole file with F5 make everything work. I guess it will be my work around for now.

Regards
